File: run_migrations.py
#!/usr/bin/env python3
"""
Script para ejecutar las migraciones de base de datos en Supabase
Ejecuta: python3 run_migrations.py
"""

import subprocess
import sys

# Leer el archivo ALL_SETUP.sql
with open('supabase/ALL_SETUP.sql', 'r') as f:
    sql_content = f.read()

# Usar supabase CLI para ejecutar el SQL
# Primero, intentamos con supabase db execute
print("🔄 Ejecutando migraciones de base de datos...")
print("=" * 60)

# Dividir el script en chunks si es necesario para evitar timeout
queries = [q.strip() for q in sql_content.split(';') if q.strip()]

print(f"Total de queries a ejecutar: {len(queries)}")
print("\n✋ IMPORTANTE: Necesitas tener logueada la sesión en Supabase CLI.")
print("Para ello, ejecuta primero: supabase login")
print("\nAlternativamente, puedes pegar manualmente el contenido de:")
print("supabase/ALL_SETUP.sql")
print("\nEn: Supabase Dashboard > SQL Editor > Nueva query")
print("=" * 60)

# Intentar ejecutar a través de la CLI (requiere estar logueado)
try:
    result = subprocess.run(
        ['supabase', 'db', 'execute', '--stdin'],
        input=sql_content.encode(),
        capture_output=True,
        text=False,
        timeout=60
    )
    
    if result.returncode == 0:
        print("✅ Migraciones ejecutadas exitosamente!")
        print(result.stdout.decode() if result.stdout else "")
    else:
        print("❌ Error al ejecutar migraciones:")
        print(result.stderr.decode() if result.stderr else "")
        print("\nIntenta hacer login primero con: supabase login")
        sys.exit(1)
        
except subprocess.TimeoutExpired:
    print("⏱️ Timeout ejecutando migraciones")
    sys.exit(1)
except FileNotFoundError:
    print("❌ supabase CLI no está instalada")
    sys.exit(1)
except Exception as e:
    print(f"❌ Error: {e}")
    sys.exit(1)

File: run_migrations.js
// Script para ejecutar migraciones en Supabase
// Ejecutar con: node run_migrations.js

import { createClient } from '@supabase/supabase-js'
import fs from 'fs'

const SUPABASE_URL = process.env.VITE_SUPABASE_URL || 'https://dayuizjuobfpzytyqcgs.supabase.co'
const SUPABASE_ANON_KEY = process.env.VITE_SUPABASE_ANON_KEY

if (!SUPABASE_URL || !SUPABASE_ANON_KEY) {
  console.error('❌ Error: VITE_SUPABASE_URL y VITE_SUPABASE_ANON_KEY no están configuradas')
  console.error('Verifica que .env.local exista con las credenciales de Supabase')
  process.exit(1)
}

console.log('🔄 Conectando a Supabase...')
console.log(`URL: ${SUPABASE_URL}`)

// La ANON_KEY no tiene permisos admin para ejecutar DDL
console.error(`❌ Error: No se puede ejecutar migraciones con ANON_KEY`)
console.error(`Necesitas un SERVICE_ROLE_KEY (token admin) para ejecutar cambios de esquema`)
console.log(`\n✋ Solución alternativa:`)
console.log(`1. Ve a Supabase Dashboard > SQL Editor`)
console.log(`2. Crea una nueva query`)
console.log(`3. Pega el contenido de: supabase/ALL_SETUP.sql`)
console.log(`4. Ejecuta la query`)
